Formula/tea.rb

```diff
@@ -0,0 +1,23 @@
+class Tea < Formula
+  desc "Command-line tool to interact with Gitea servers"
+  homepage "https://gitea.com/gitea/tea"
+  url "https://gitea.com/gitea/tea/archive/v0.8.0.tar.gz"
+  sha256 "6c73c0a7b66cdfd1e5a302257d54df06a3a41eb9bdbfeb547966db431ae23b23"
+  license "MIT"
+  head "https://gitea.com/gitea/tea.git", branch: "master"
+
+  depends_on "go" => :build
+
+  def install
+    # Remove `if build.head?` on next release.
+    system "make", "vendor" if build.head?
+    system "make"
+    bin.install "tea"
+    bash_completion.install "contrib/autocomplete.sh" => "tea"
+    zsh_completion.install "contrib/autocomplete.zsh" => "_tea"
+  end
+
+  test do
+    system "#{bin}/tea", "-h"
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
I absolutely understand that, but I failed to find any functionality that didn't require login.

Reply:
> If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.

Sure, but how do I confirm that a command is failing? I searched and couldn't find how to do that.

Reply:
`shell_output` takes the expected exit code as a second argument.